stmhal/boards/ESPRUINO_PICO/mpconfigboard.h

```diff
@@ -2,6 +2,12 @@
 #define MICROPY_HW_MCU_NAME         "STM32F401CD"
 #define MICROPY_PY_SYS_PLATFORM     "pyboard"
 
+#undef  MICROPY_PY_USOCKET
```

Comment:
I think it would just move the #undefs (or we'd need to add #ifndef) into mpconfigport.h. Using #ifndef would consistent with what mpconfig.h does.